spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java

```diff
@@ -21,165 +21,122 @@
-    private String adminResourceToSearchForApiKeys;
-
-    @Value("${ygg.admin.resources.search-service-by-api-key-value}")
-    private String adminResourceToGetServiceByApiKeyValue;
+        authoritiesGrantedToYggAdmin.add(new SimpleGrantedAuthority(
+                ServiceApiAccessRightsVoter.ACCESS_SERVICE_AUTHORITY_PREFIX + "refresh"));
+    }
 
     @Override
     public Authentication authenticate(Authentication authentication) throws AuthenticationException {
-
         try {
-            Assert.notNull(authentication.getCredentials());
-            Assert.isInstanceOf(String.class, authentication.getCredentials());
-            String apiKey = (String) authentication.getCredentials();
-            Assert.hasText(apiKey);
-
-            //TODO this is just a workaround for now: check for fixed API key to allow CoatRack admin access
-            if (apiKey.equals(ApiKey.API_KEY_FOR_YGG_ADMIN_TO_ACCESS_PROXIES)) {
-
-                Set<SimpleGrantedAuthority> authoritiesGrantedToYggAdmin = new HashSet<>();
-                authoritiesGrantedToYggAdmin.add(new SimpleGrantedAuthority(ServiceApiAccessRightsVoter.ACCESS_SERVICE_AUTHORITY_PREFIX + "refresh"));
-
-                ApiKeyAuthToken apiKeyAuthToken = new ApiKeyAuthToken(apiKey, authoritiesGrantedToYggAdmin);
-                apiKeyAuthToken.setAuthenticated(true);
-                return apiKeyAuthToken;
-            }
-
-            // regular api consumer's api key check
-            if (isApiKeyValid(apiKey)) {
-                // key is valid, now get service api URI identifier
-                ServiceApi serviceApi = getServiceApiByApiKey(apiKey);
-                String uriIdentifier = serviceApi.getUriIdentifier();
-
-                // add uri identifier as granted authority
-                ApiKeyAuthToken apiKeyAuthToken = new ApiKeyAuthToken(apiKey, Collections.singleton(
-                        new SimpleGrantedAuthority(
-                                ServiceApiAccessRightsVoter.ACCESS_SERVICE_AUTHORITY_PREFIX + uriIdentifier)));
-
-                apiKeyAuthToken.setAuthenticated(true);
-                return apiKeyAuthToken;
-            } else {
-                // TODO log this via MetricsCounterService?
-                throw new BadCredentialsException("API key value is not valid");
-            }
-        } catch (IllegalArgumentException e) {
-            log.debug("API key parameter missing or invalid: " + e.getMessage());
-            throw new BadCredentialsException("API key parameter missing or invalid: " + e.getMessage());
+            log.debug("Verifying the authentication {}.", authentication.getName());
+            return createApiKeyAuthToken(authentication);
+        } catch (Exception e) {
+            log.debug("During the authentication process this exception occurred: ", e);
         }
+        throw new SessionAuthenticationException("The authentication process failed.");
```

Comment:
For now I replaced that with a BadCredentialsException. However, I suggested in an other comment to replace this by 'AuthenticationProcessFailedException' because bad credentials are not the only reason for an authentication to fail.